### PR TITLE
[wni] Make azure WNI use the given credential file

### PR DESCRIPTION
**Pull request by sebsoto**
This PR makes the azure WNI API use the credentials passed to it

### CODE REVIEW

File: tools/windows-node-installer/test/e2e/azure_test.go [hunk 1]
```diff
@@ -107,20 +112,43 @@ var (
 	credentials *types.Credentials
 )
 
+// getAzureCloudProvider returns an azure cloud provider
+func getAzureCloudProvider() (cloudprovider.Cloud, error) {
+	cloud, err := cloudprovider.CloudProviderFactory(kubeconfig, azureCredentials, "", artifactDir, azureImageID,
```
Comment (VaishnaviHire) [inline tools/windows-node-installer/test/e2e/azure_test.go:117]: We should not use `CloudProviderFactory` method here, as we are using service account to create oc client for CI. We need to do it similar to [aws_test](https://github.com/openshift/windows-machine-config-bootstrapper/blob/master/tools/windows-node-installer/test/e2e/aws_test.go#L111)

Comment (sebsoto) [reply tools/windows-node-installer/test/e2e/azure_test.go:117]: I agree that we should be using the service account to create the oc client for CI.

However the actual function that needs to be tested is `cloudprovider.CloudProviderFactory` as that is what we use in the operator. Directly calling the azure cloud provider is skipping testing a portion of the code.

My intention here was to change as little as possible with the end to end tests to make them actually work. Previously we were not using the service account to create an oc client, I feel that adding that is outside the scope of this ticket.

File: tools/windows-node-installer/pkg/cloudprovider/azure/vm.go [hunk 2]
```diff
@@ -870,10 +878,10 @@ func (az *AzureProvider) constructNetworkProfile(ctx context.Context,
 		ipConfigName := az.generateResourceName("ipConfig", vmRandString)
 		az.NicName = nicName
 		ptrvmNic, err := az.createNIC(ctx, vnetName, subnetName, nsgName, ipConfigName)
```
Comment (mansikulkarni96) [inline tools/windows-node-installer/pkg/cloudprovider/azure/vm.go:880]: ptrvmNic -> can this be renamed to something that is more understandable and follows go naming conventions?

File: tools/windows-node-installer/test/e2e/azure_test.go [hunk 3]
```diff
@@ -107,20 +112,43 @@ var (
...
-// TODO: This is not actually testing the Windows VM creation. Change this function.
 func TestCreateVM(t *testing.T) {
-	err := setup()
-	require.NoErrorf(t, err, "failed at the setup with error: %v", err)
+	cloud, err := getAzureCloudProvider()
+	require.NoError(t, err, "error getting azure cloud provider")
+	vm, err := cloud.CreateWindowsVM()
+	require.NoError(t, err, "error creating Windows VM")
+
+	// TODO: This needs to be removed and we should be using the vm returned by CreateWindowsVM() to test
```
Comment (mansikulkarni96) [inline tools/windows-node-installer/test/e2e/azure_test.go:139]: do we have a story to track this TODO?

File: tools/windows-node-installer/test/e2e/aws_test.go [hunk 4]
```diff
@@ -16,22 +16,13 @@ import (
...
-	imageID      = ""
-	instanceType = "m4.large"
-	sshKey       = "libra"
-
 	// awsProvider is setup as a variable for both creating, destroying,
 	// and tear down Windows instance in case test fails in the middle.
 	awsProvider = &awscp.AwsProvider{}
 
+	// awscredentials is set by OpenShift CI
+	awscredentials = os.Getenv("AWS_SHARED_CREDENTIALS_FILE")
```
Comment (mansikulkarni96) [inline tools/windows-node-installer/test/e2e/aws_test.go:24]: please rename to awsCredentials

File: tools/windows-node-installer/test/e2e/azure_test.go [hunk 5]
```diff
@@ -107,20 +112,43 @@ var (
...
+// getAzureCloudProvider returns an azure cloud provider
+func getAzureCloudProvider() (cloudprovider.Cloud, error) {
+	cloud, err := cloudprovider.CloudProviderFactory(kubeconfig, azureCredentials, "", artifactDir, azureImageID,
+		azureInstanceType, sshKey, privateKeyPath)
+	if err != nil {
+		return nil, errors.Wrap(err, "error creating cloud provider")
+	}
+	_, ok := cloud.(*wniAzure.AzureProvider)
+	if !ok {
+		return nil, fmt.Errorf("did not get an AzureProvider from cloud factory")
```
Comment (mansikulkarni96) [inline tools/windows-node-installer/test/e2e/azure_test.go:124]: can reframe to fmt.Errorf("could not get an AzureProvider from cloud factory")

File: tools/windows-node-installer/pkg/cloudprovider/azure/vm.go (hunk 2 above)
Comment (aravindhp) [reply tools/windows-node-installer/pkg/cloudprovider/azure/vm.go:880]: Given the amount of technical debt we have in this repo, I am fine changing things when are making changes in the area. This of course is withing reason :-)

File: tools/windows-node-installer/pkg/cloudprovider/azure/vm.go [hunk 6]
```diff
@@ -870,10 +870,10 @@ func (az *AzureProvider) constructNetworkProfile(ctx context.Context,
...
-		vmNic = *(ptrvmNic)
-		if errorCheck(err) {
+		if err != nil {
 			return nil, fmt.Errorf("failed to create nic for the instance: %v", err)
 		}
+		vmNic = *(ptrvmNic)
```
Comment (aravindhp) [inline tools/windows-node-installer/pkg/cloudprovider/azure/vm.go:884]: Please remove these `()`

File: tools/windows-node-installer/test/e2e/azure_test.go [hunk 7]
```diff
@@ -151,11 +179,11 @@ func constructRequiredRules() (map[string]*requiredRule,
 	return requiredRules, nil
 }
 
-// setup does these prerequisite tests before running the tests.
+// loadInstanceInfo does these prerequisite tests before running the tests.
```
Comment (aravindhp) [inline tools/windows-node-installer/test/e2e/azure_test.go:182]: Isn't this doing more than just loading instance information? Not sure if this is the correct name.

File: tools/windows-node-installer/test/e2e/azure_test.go (hunk 3 above)
Comment (sebsoto) [reply tools/windows-node-installer/test/e2e/azure_test.go:139]: Added

**Comment by mansikulkarni96** [comment]
LGTM

File: tools/windows-node-installer/test/e2e/common.go [hunk 8]
```diff
@@ -27,6 +28,16 @@ var (
...
 	retryInterval = 5 * time.Second
+
+	kubeconfig     = os.Getenv("KUBECONFIG")
+	artifactDir    = os.Getenv("ARTIFACT_DIR")
+	privateKeyPath = os.Getenv("KUBE_SSH_KEY_PATH")
+
+	// imageID is the image that will be fed to the WNI for the tests. This is being set to empty, as we wish for it
+	// to use the latest Windows image
+	imageID      = ""
+	instanceType = "m4.large"
```
Comment (aravindhp) [inline tools/windows-node-installer/test/e2e/common.go:39]: This is not common but AWS specific.

File: tools/windows-node-installer/pkg/cloudprovider/azure/vm.go [hunk 9]
```diff
@@ -859,23 +859,23 @@ func (az *AzureProvider) constructNetworkProfile(ctx context.Context,
 		}
 	}
 
-	var vmNic network.Interface
+	var nicID *string
```
Comment (aravindhp) [inline tools/windows-node-installer/pkg/cloudprovider/azure/vm.go:870]: I would remove this variable and use `nic.ID` directly. But add a check to ensure `nic.ID` is not `nil`

File: Makefile [hunk 10]
```diff
@@ -42,6 +42,10 @@ test-unit-wni:
 test-e2e-tools:
 	cd ./tools/windows-node-installer && $(GO_BUILD_ARGS) go test -run=TestAwsE2eSerial $(TOOLS_DIR)/test/e2e/... -timeout 20m -v
 
+.PHONY: test-e2e-tools-azure
```
Comment (aravindhp) [inline Makefile:45]: `test-e2e-wni-azure`. Let us move away from the previous naming convention.

**Pull request merged**